src/parser/jobs/pct/modules/Gauge.tsx

```diff
@@ -0,0 +1,367 @@
+		} else if (event.type === 'action') {
+			eventActionId = event.action
+		}
+		if (eventActionId == null) { return }
+
+		// Subtractive Pallet does not spend gauge if Subtractive Spectrum is currently active
+		if (eventActionId === this.data.actions.SUBTRACTIVE_PALLETTE.id &&
+			this.actors.current.hasStatus(this.data.statuses.SUBTRACTIVE_SPECTRUM.id)) {
+			return
+		}
+
+		const modifier = this.paletteGaugeModifiers.get(eventActionId)
+		if (modifier == null) { return }
+
+		this.paletteGauge.modify(modifier[event.type] ?? 0)
+	}
+
+	private onWhitePaintGenerator(event: Events['damage']) {
+		if (!isSuccessfulHit(event)) { return }
+
+		// If we need to generate a Black Paint because there were no White Paints available when the player hit Subtractive Palette,
+		// do that instead
+		if (this.nextWhiteIsBlack) {
+			this.paintGauge.generate(BLACK_PAINT)
+			this.nextWhiteIsBlack = false
+			return
+		}
+
+		// Otherwise generate the White Paint as expected
+		this.paintGauge.generate(WHITE_PAINT)
+	}
+
+	private onWhitePaintSpender() {
+		this.paintGauge.spend(WHITE_PAINT)
+	}
+
+	private onBlackPaintSpender() {
+		this.paintGauge.spend(BLACK_PAINT)
+	}
+
+	private onSubtractivePalette(event: Events['action']) {
+		// If we have a White Paint available, Subtractive Palette swaps it for a Black Paint
+		if (this.paintGauge.getCountAt(WHITE_PAINT, event.timestamp) > 0) {
+			this.paintGauge.spend(WHITE_PAINT)
+			this.paintGauge.generate(BLACK_PAINT)
+			return
+		}
+
+		// Otherwise, the next White Paint that would generate will be a Black Paint instead
+		this.nextWhiteIsBlack = true
+	}
+
+	private onCanvasModifier(event: Events['action']) {
+		const modifier = this.canvasModifiers.get(event.action)
+		if (modifier == null) { return }
+
+		const canvasMod = modifier[event.type]
+		if (canvasMod == null) { return }
+
+		if (canvasMod.type === 'generate') {
+			this.canvasGauge.generate(canvasMod.which)
+		} else if (canvasMod.type === 'spend') {
+			this.canvasGauge.spend(canvasMod.which)
+		}
+	}
+
+	private onDepictionModifier(event: Events['action']) {
+		switch (event.action) {
+		// Fanged Muse resets the depiction state while generating Madeen
+		case this.data.actions.FANGED_MUSE.id:
+			this.depictionGauge.reset()
+			break
+		// The other muses add their creature to the depiction set
+		case this.data.actions.CLAWED_MUSE.id:
+			this.depictionGauge.generate(CLAW_MOTIF)
+		// Switch case fallthrough as a pseudo history correction for dungeon/24man depiction state carryover since this isn't a counter gauge with correction built in
+		/* eslint-disable no-fallthrough */
+		case this.data.actions.WINGED_MUSE.id:
+			if (!this.depictionGauge.getStateAt(WING_MOTIF)) {
+				this.depictionGauge.generate(WING_MOTIF)
+			}
+		case this.data.actions.POM_MUSE.id:
+			if (!this.depictionGauge.getStateAt(POM_MOTIF)) {
+				this.depictionGauge.generate(POM_MOTIF)
+			}
+			/* eslint-enable no-fallthrough */
+		}
```

Comment:
Shouldn't we be able to generalize this history correction so that we can make it available to all SetGauge consumers?  SAM stickers are going to have a similar problem when the user uses Midare after only one sticker is generated in a pull.

Reply:
Theoretically yes, but I'd like to do a separate PR to add core history correction to SetGauge, rather than try to stuff it onto this one.